PWGHF/Utils/utilsPid.h

```
@@ -0,0 +1,108 @@
+{
+  static constexpr float defaultNSigmaTolerance = .1f;
+  static constexpr float defaultNSigma = -999.f + defaultNSigmaTolerance; // -999.f is the default value set in TPCPIDResponse.h and PIDTOF.h
+
+  if constexpr (tiny) {
+    tpcNSigma *= aod::pidtpc_tiny::binning::bin_width;
+    tofNSigma *= aod::pidtof_tiny::binning::bin_width;
+  }
+
+  if ((tpcNSigma > defaultNSigma) && (tofNSigma > defaultNSigma)) { // TPC and TOF
+    return std::sqrt(.5f * (tpcNSigma * tpcNSigma + tofNSigma * tofNSigma));
+  }
+  if (tpcNSigma > defaultNSigma) { // only TPC
+    return std::abs(tpcNSigma);
+  }
+  if (tofNSigma > defaultNSigma) { // only TOF
+    return std::abs(tofNSigma);
+  }
+  return tofNSigma; // no TPC nor TOF
+}
+
+/// @brief Function to fill tables with HF prong PID information
+/// @tparam TRK datatype of the prong track
+/// @tparam ROW datatype of the prong PID table to fill
+/// @tparam specPid particle species
+/// @param track prong track
+/// @param rowPid cursor of the prong PID table to fill
+template <int specPid, typename TRK, typename ROW>
+void fillProngPid(TRK const& track, ROW& rowPid)
+{
+
+  // get PID information for the daughter tracks
+  // TODO: add here the code for a possible PID post-calibrations in MC
+  float nSigTpc = -999.f;
+  float nSigTof = -999.f;
+  if constexpr (specPid == HfProngSpecies::Pion) {
+    // pion PID
+    if (track.hasTPC()) {
+      nSigTpc = track.tpcNSigmaPi();
+    }
+    if (track.hasTOF()) {
+      nSigTof = track.tofNSigmaPi();
+    }
+  } else if constexpr (specPid == HfProngSpecies::Kaon) {
+    // kaon PID
+    if (track.hasTPC()) {
+      nSigTpc = track.tpcNSigmaKa();
+    }
+    if (track.hasTOF()) {
+      nSigTof = track.tofNSigmaKa();
+    }
+  } else if constexpr (specPid == HfProngSpecies::Proton) {
+    // proton PID
+    if (track.hasTPC()) {
+      nSigTpc = track.tpcNSigmaPr();
+    }
+    if (track.hasTOF()) {
+      nSigTof = track.tofNSigmaPr();
+    }
+  }
+
```

I would add an `else` clause here and throw an exception or a fatal error for unsupported PIDs.